app/e2e/outgoingcall.test.js

```diff
@@ -0,0 +1,82 @@
+    await element(by.id('login_button')).tap();
+    await element(by.id('dialer_button')).tap();
+  };
+
+  beforeEach(async () => {
+    await global.device.reloadReactNative();
+    await navigateToDialer();
+  });
+
+  it('should allow user to enter PTSN number', async () => {
+    await element(by.id('dialpad_button_1')).tap();
+    await element(by.id('dialpad_button_2')).tap();
+    await element(by.id('dialpad_button_3')).tap();
+    await element(by.id('dialpad_button_4')).tap();
+    await element(by.id('dialpad_button_4')).tap();
+    await element(by.id('dialpad_button_4')).tap();
+    await element(by.id('dialpad_button_8')).tap();
+    await element(by.id('dialpad_button_8')).tap();
+    await element(by.id('dialpad_button_8')).tap();
+    await element(by.id('dialpad_button_8')).tap();
+    await expect(element(by.id('formatted_number'))).toHaveText('+1234448888');
+  });
+
+  it('should allow user to enter Client-ID', async () => {
+    await element(by.text('Client')).tap();
+    await element(by.id('client_text_input')).typeText('Client-Web');
+    await expect(element(by.id('client_text_input'))).toHaveText('Client-Web');
+  });
+
+  if (device.getPlatform() === 'android') {
+    describe('Android: successful connect', () => {
+      it('should make a successful PTSN call', async () => {
+        await element(by.id('dialpad_button_3')).tap();
+        await element(by.id('dialpad_button_1')).tap();
+        await element(by.id('dialpad_button_5')).tap();
+        await element(by.id('dialpad_button_6')).tap();
+        await element(by.id('dialpad_button_7')).tap();
+        await element(by.id('dialpad_button_0')).tap();
+        await element(by.id('dialpad_button_3')).tap();
+        await element(by.id('dialpad_button_5')).tap();
+        await element(by.id('dialpad_button_4')).tap();
+        await element(by.id('dialpad_button_1')).tap();
+        await element(by.id('call_button')).tap();
+        await waitFor(element(by.id('active_call'))).toBeVisible();
+        await waitFor(element(by.id('call_status'))).toHaveText('ringing');
+        await waitFor(element(by.id('call_status'))).toHaveText('00:05');
+        await waitFor(element(by.id('active_call'))).not.toBeVisible();
+      });
+    });
+
+    describe('disconnect', () => {
+      it('should disconnect if invalid number', async () => {
+        await element(by.id('dialpad_button_1')).tap();
+        await element(by.id('dialpad_button_2')).tap();
+        await element(by.id('dialpad_button_3')).tap();
+        await element(by.id('call_button')).tap();
+        await waitFor(element(by.id('active_call'))).toBeVisible();
+        await waitFor(element(by.id('call_status'))).toHaveText('ringing');
+        await waitFor(element(by.id('call_status'))).toHaveText('disconnected');
+        await waitFor(element(by.id('active_call'))).not.toBeVisible();
+      });
+
+      it('should disconnect if invalid Client-ID', async () => {
```

Comment:
Same here, we should check that the call has been disconnected right away.

Reply:
Disconnect still doesn't happen right away. We still have the user see texts `ringinging` -> `disconnect` -> back to dial screen.